grove/grovetccfg/grovetccfg.go

```diff
@@ -658,10 +658,10 @@ func makeServersHostnameMap(servers []tc.Server) map[string]tc.Server {
 	return m
 }
 
-func makeCachegroupsNameMap(cgs []tc.CacheGroup) map[string]tc.CacheGroup {
-	m := map[string]tc.CacheGroup{}
+func makeCachegroupsNameMap(cgs []tc.CacheGroupNullable) map[string]tc.CacheGroupNullable {
+	m := map[string]tc.CacheGroupNullable{}
 	for _, cg := range cgs {
-		m[cg.Name] = cg
+		m[*cg.Name] = cg
```

Comment:
need to check for `nil` names now that we're using nullables, this could segfault